allure-support/src/main/kotlin/com/kaspersky/components/alluresupport/files/dirs/AllureDirsProvider.kt

```diff
@@ -0,0 +1,56 @@
+package com.kaspersky.components.alluresupport.files.dirs
+
+import android.annotation.SuppressLint
+import android.app.Instrumentation
+import android.content.Context
+import android.os.Build
+import android.os.Environment
+import com.kaspersky.kaspresso.files.dirs.DefaultDirsProvider
+import com.kaspersky.kaspresso.files.resources.ResourcesRootDirsProvider
+import com.kaspersky.kaspresso.instrumental.InstrumentalDependencyProvider
+import com.kaspersky.kaspresso.internal.extensions.other.createDirIfNeeded
+import com.kaspersky.kaspresso.internal.extensions.other.createFileIfNeeded
+import java.io.File
+
+class AllureDirsProvider(
+    private val instrumentation: Instrumentation,
+    private val resourcesRootDirsProvider: ResourcesRootDirsProvider,
+    instrumentationDependencyProvider: InstrumentalDependencyProvider
+) : DefaultDirsProvider(instrumentationDependencyProvider) {
+
+    @Suppress("DEPRECATION")
+    @SuppressLint("WorldReadableFiles", "ObsoleteSdkInt")
+    override fun provideNew(dest: File): File {
```

Comment:
Maybe it would be better to declare explicitly two methods for two types of dirs providing? And also name them in a way excluding video/non-video words.

Reply:
I started it, but it requires much more code changes that I'd like (DirsProvider and ResourcesDirsProvider imeplemtations). I think for now it's better to leave it as it is and do a refactor later.